File: test_imdb.js
const https = require('https');

https.get('https://www.imdb.com/title/tt0816692/', {
  headers: {
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36'
  }
}, (res) => {
  let data = '';
  res.on('data', chunk => data += chunk);
  res.on('end', () => {
    const match = data.match(/playbackURLs":\[(.*?)]/);
    if (match) {
      console.log('Found playback URLs!');
      console.log(match[0].substring(0, 500));
    } else {
      console.log('No playback URLs found.');
      // Maybe check for data-video id?
      const videoIdMatch = data.match(/vi\d+/g);
      if (videoIdMatch) {
         console.log('Found video IDs:', [...new Set(videoIdMatch)]);
      }
    }
  });
}).on('error', err => console.log(err));

File: test_imdb.py
import urllib.request
import re

url = 'https://www.imdb.com/title/tt0816692/'
req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64)'})
try:
    with urllib.request.urlopen(req) as response:
        html = response.read().decode('utf-8')
        matches = re.findall(r'playbackURLs":\[(.*?)\]', html)
        if matches:
            print('Found playback URLs!')
            print(matches[0][:500])
        else:
            print('No playback URLs found.')
            video_ids = set(re.findall(r'vi\d+', html))
            print('Found video IDs:', list(video_ids)[:10])
except Exception as e:
    print(e)
